octoprint_firmwareupdate/__init__.py

```diff
@@ -256,8 +253,28 @@ def _update_from_github(self):
         # Write version to File
         with open(self.version_file, 'w') as f:
             f.write(rjson['assets'][0]['updated_at'])
-        urllib.urlretrieve(rjson['assets'][0][
-                           'browser_download_url'], self.firmware_file)
+
+        # Download the hex file from GitHub
+        try:
+            r = requests.get(rjson['assets'][0]['browser_download_url'],
+                             stream=True, timeout=3.05)
+            r.raise_for_status()
+        except (requests.exceptions.ConnectionError,
+                requests.exceptions.HTTPError) as e:
+            self.raise_connection_error(e)
+            return
+
+        with open(self.firmware_file, 'wb') as f:
+            try:
+                for chunk in r.iter_content(chunk_size=8192):
+                    if not chunk:
```

Comment:
this is done to filter out keep-alive new chunks, however, i think i'm going to revert to using the raw response.